Add white stroke to realm icon

FadhlanR opened the pull request:
![Screenshot 2023-09-14 at 19 00 16](https://github.com/cardstack/boxel/assets/12637010/c9bc437e-83c2-4f60-8fa9-f0c4cae29ae2)
![Screenshot 2023-09-14 at 19 00 10](https://github.com/cardstack/boxel/assets/12637010/b3961bf1-81f6-476d-bafc-795a595cb4d5)

FadhlanR committed:
Add white stroke to realm icon

jurgenwerk reviewed (changes requested):
![image](https://github.com/cardstack/boxel/assets/273660/5841e195-a0b1-4e96-ba13-12c39c203d41)
![image](https://github.com/cardstack/boxel/assets/273660/0c7bf92e-fe82-4ed4-82e3-d0b9eb31ba72)

This looks weird and blurry. I don't think it was meant to be done like this.

FadhlanR commented:
> ![image](https://github.com/cardstack/boxel/assets/273660/5841e195-a0b1-4e96-ba13-12c39c203d41)
> 
> ![image](https://github.com/cardstack/boxel/assets/273660/0c7bf92e-fe82-4ed4-82e3-d0b9eb31ba72)
> 
> 
> 
> This looks weird and blurry. I don't think it was meant to be done like this. 


I've tried using border/outline but it didn't solve the problem since the border/outline will be shown as white rectangle and the background will still be black.

burieberry reviewed (changes requested):
I think what we want is a transparent square with slightly rounded white border. We can set the logo as background-image or place it inside this container

FadhlanR committed:
Add background

FadhlanR commented:
@jurgenwerk @burieberry After implementing the background, does it look like what we expected?


![Screenshot 2023-09-15 at 12 58 34](https://github.com/cardstack/boxel/assets/12637010/996aaf08-bf3f-4345-8555-b3a3fc097436)
![Screenshot 2023-09-15 at 12 58 45](https://github.com/cardstack/boxel/assets/12637010/85ee49fc-ae59-4358-a5ce-0530061adb0d)

burieberry commented:
> @jurgenwerk @burieberry After implementing the background, does it look like what we expected?
> 
> ![Screenshot 2023-09-15 at 12 58 34](https://user-images.githubusercontent.com/12637010/268188714-996aaf08-bf3f-4345-8555-b3a3fc097436.png) ![Screenshot 2023-09-15 at 12 58 45](https://user-images.githubusercontent.com/12637010/268188720-85ee49fc-ae59-4358-a5ce-0530061adb0d.png)

I don't know about using the realm background. Maybe @lukemelia know what the logo should look like? I see what you mean about the white border/outline, though.

FadhlanR committed:
Merge branch 'main' into cs-5955-add-stroke-to-realm-icon

FadhlanR committed:
Update border rules

FadhlanR commented:
I have updated this PR based on the new design guide from @cmgardella .

Designs:
![30x30_realm_icon__on_white_background__non_transparent](https://github.com/cardstack/boxel/assets/12637010/5ccdbc13-00d4-478b-9a8a-92bf46d5a831)
![18x18_realm_icon__on_dark_background__non_transparent](https://github.com/cardstack/boxel/assets/12637010/6fe93a7e-0088-4df5-be48-cbc773e2c723)
![18x18_realm_icon__on_dark_background__transparent_image](https://github.com/cardstack/boxel/assets/12637010/c4dd1a88-0246-49f0-9219-88b9fcd17d13)

Implementations:
![Screenshot 2023-09-21 at 18 11 39](https://github.com/cardstack/boxel/assets/12637010/94ea2ab1-6443-4f42-ac19-7d1723df8772)
![Screenshot 2023-09-21 at 18 13 18](https://github.com/cardstack/boxel/assets/12637010/b45d8215-4c84-41d3-8b8a-bae9c349d71a)
![Screenshot 2023-09-21 at 18 11 00](https://github.com/cardstack/boxel/assets/12637010/ab79e114-4d78-4989-a870-a9211e739949)
![Screenshot 2023-09-21 at 18 11 09](https://github.com/cardstack/boxel/assets/12637010/851a94f1-0fc5-4447-87ac-82becbba38d3)

For an icon with a transparent background on a dark background (card URL bar), we added padding and a white background to it.

FadhlanR committed:
test fixes

jurgenwerk commented on packages/host/package.json:
```diff
@@ -176,6 +176,7 @@
     "ember-async-data": "^1.0.1",
-    "ember-click-outside": "^6.0.1"
+    "ember-click-outside": "^6.0.1",
+    "jimp": "^0.22.10"
```
jimp's unpacked size is 4.96 MB 😬

FadhlanR replied:
Hmm, I believe we don't need to check whether the icon has a transparent background. Instead, we can simply place a white background behind the icon. If the icon already has a solid background, it will effectively cover the white background. What do you think?

lukemelia replied:
Yes, the intention is always use a white background behind any realm icon.

FadhlanR committed:
Merge branch 'main' into cs-5955-add-stroke-to-realm-icon

lukemelia commented:
I don't think there should be such a prominent black rule around the icon when on white background

FadhlanR committed:
Simplify the logic

FadhlanR committed:
Fix border icon in stack item

jurgenwerk commented:
Can you show how it looks now?

FadhlanR committed:
Update border radius size

FadhlanR commented:
> Can you show how it looks now?
![Screenshot 2023-09-22 at 14 13 43](https://github.com/cardstack/boxel/assets/12637010/f5dbbbc1-52a4-440c-8fda-a7d2b298d931)
![Screenshot 2023-09-22 at 14 13 36](https://github.com/cardstack/boxel/assets/12637010/033b81e2-7264-43d7-85fc-54726cf81377)

jurgenwerk commented:
@FadhlanR in your 2 images above, in one case the realm icon container is square and in the other case it's not. I think it would be good to make this consistent.

FadhlanR committed:
Make the shape consistent

FadhlanR commented:
> @FadhlanR in your 2 images above, in one case the realm icon container is square and in the other case it's not. I think it would be good to make this consistent.

Updated.
<img width="240" alt="Screenshot 2023-09-26 at 14 44 10" src="https://github.com/cardstack/boxel/assets/12637010/4b2bc94d-8d79-43cd-a5d4-d878bc4dfd1d">
<img width="189" alt="Screenshot 2023-09-26 at 14 44 03" src="https://github.com/cardstack/boxel/assets/12637010/0e1be847-846c-4b54-a9b6-4938c0a53c74">
